cmake/modules/RootBuildOptions.cmake

```diff
@@ -442,3 +442,12 @@ if(macos_native)
     message(STATUS "Option 'macos_native' is only for MacOS systems. Ignoring it.")
   endif()
 endif()
+
+# Print message saying with which versions of Python are used to build
+if(pyroot_experimental)
```

Comment:
I understand we moved this here just to have access to the value of `pyroot_experimental`, but we left in `SearchRootCoreDeps` the equivalent message for the one-python option. Two comments:

1. Shouldn't we have all these messages in the same place?
2. Where is the right place to have these messages?

Reply:
@oshadura

Reply:
@etejedor SearchRootCoreDeps is included *before* RootOption.cmake. Any conditional statement using ROOT options used here will be just ignored.

Reply:
Also:

3.  In old PyROOT, we will not print this message. But we will search for Python2 and Python3, so users will see that we found Python2 and Python3 (from `find_package`), but they will not see any message regarding which python version ROOT will be built with. As things are now in this PR, they will only see such message if they specified `PYTHON_EXECUTABLE`.

Reply:
Note: SearchRootCoreDeps.cmake was created with purpose to be generic for whole ROOT. It reperesent **core dependencies** of ROOT without which we can't build ROOT.

Reply:
The right place for this will be root build options where we define or option setup or and limitations/exceptions (see example of c++ modules) and etc.

Reply:
@oshadura I understand that your answer to 2) is `RootBuildOptions.cmake`. What is the answer for 1) and 3)?

Reply:
Are you taking about https://github.com/root-project/root/blob/master/cmake/modules/SearchRootCoreDeps.cmake#L11 here is mini hack from Massimiliano (it will be needed only if you change to the new configuration, but in the old build directory already configured with different configuration). @maxgalli do we still need this hack? (I suppose yes, can you add a comment so there will be no confusion next time)

Reply:
> Also:
> 
>     1. In old PyROOT, we will not print this message. But we will search for Python2 and Python3, so users will see that we found Python2 and Python3 (from `find_package`), but they will not see any message regarding which python version ROOT will be built with. As things are now in this PR, they will only see such message if they specified `PYTHON_EXECUTABLE`.

Here I am not sure what to reply...It is up to you! (once we will need to make some order with messages :) )

Reply:
for what concerns the hack I already got rid of it because it's not needed anymore

Reply:
No I was not talking about that code. I asked 1) because, when the user specifies `PYTHON_EXECUTABLE`, the message is printed (for both new PyROOT and old) in `SearchRootCoreDeps`, here:

https://github.com/root-project/root/pull/5496/files#diff-5a4af40c5dab2eba7ad500714197987aR42

but the equivalent messages when we are dealing with Python2 and Python3 are printed in `RootBuildOptions.cmake`. It just seems to me that we took a piece of the logic that was in `SearchRootCoreDeps` and put it `RootBuildOptions` just because we needed the `pyroot_experimental` variable, but we left behind equivalent code in `SearchRootCoreDeps`.

Reply:
I don't really see a problem, is this print correct? The point is that people who *doesn't* have pyroot=ON and pyroot_experimental=ON, they will not get these messages.

Reply:
you see the point, that we consider python interpreter as a *must* dependency, that should be present even without pyroot and pyroot_experimental (in this file (SearchRootCoreDeps.cmake) I expect we will have LLVM as well )

Reply:
We had a call with @maxgalli about how to print the messages and what to print both for PyROOT new and old. Please do a final review when he pushes the changes and we should be good to go, thanks!